### ISSUE TITLE
google-api-client requires Ruby version ~> 2.4. The current ruby version is 3.0.0.dev.

### ISSUE THREAD

**Issue by yahonda** [googleapis/google-api-ruby-client#901]
#### Environment details

  - OS: Ubuntu 20.04.1 LTS
  - Ruby version: ruby 3.0.0dev (2020-09-01T10:03:41Z master 8d1de3154c) [x86_64-linux]
  - Gem name and version: google-api-client 0.9.4.

#### Steps to reproduce

1. Install ruby 3.0.0dev using rbenv and ruby-build
2. gem install google-api-client

#### Actual results

```ruby
$ ruby -v
ruby 3.0.0dev (2020-09-01T10:03:41Z master 8d1de3154c) [x86_64-linux]
$ gem -v
3.2.0.rc.1
$ gem install google-api-client
Fetching uber-0.1.0.gem
Fetching declarative-0.0.20.gem
Fetching representable-3.0.4.gem
Fetching declarative-option-0.1.0.gem
Fetching retriable-3.1.2.gem
Fetching mini_mime-1.0.2.gem
Fetching jwt-2.2.2.gem
Fetching multi_json-1.15.0.gem
Fetching google-api-client-0.44.0.gem
Fetching signet-0.14.0.gem
Fetching memoist-0.16.2.gem
Fetching os-1.1.1.gem
Fetching googleauth-0.13.1.gem
Fetching httpclient-2.8.3.gem
Successfully installed declarative-option-0.1.0
Successfully installed declarative-0.0.20
Successfully installed uber-0.1.0
Successfully installed representable-3.0.4
Successfully installed retriable-3.1.2
Successfully installed mini_mime-1.0.2
Successfully installed jwt-2.2.2
Successfully installed multi_json-1.15.0
Successfully installed signet-0.14.0
Successfully installed memoist-0.16.2
Successfully installed os-1.1.1
Successfully installed googleauth-0.13.1
Successfully installed httpclient-2.8.3
ERROR:  Error installing google-api-client:
	The last version of google-api-client (>= 0) to support your Ruby & RubyGems was 0.9.4. Try installing it with `gem install google-api-client -v 0.9.4`
	google-api-client requires Ruby version ~> 2.4. The current ruby version is 3.0.0.dev.
$
```

#### Expected results

I'd like to install google-api-client with Ruby 3. 

#### Additional information
This line restricts Ruby versions.

https://github.com/googleapis/google-api-ruby-client/blob/cb9bd92cd1560f621cc27407e3d10e307970cf56/google-api-client.gemspec#L24

**Comment by yahonda** [comment]
Fixed by #900

**Comment by kpaxradio** [comment]
I have a same message:

> google-api-client-0.23.4 requires ruby version ~> 2.0, which is incompatible
> with the current version, ruby 3.1.2p20

in an installation, can you know what to do to fix it?

with 
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux-gnu]
and gem version:
3.3.15
